### ISSUE TITLE
Chef fails to install gems when no version is specified, but options are

### ISSUE THREAD

**Issue by tas50** [chef/chef#2551]
### Version:
11.16.4 and 12.0.0

### Environment: 
Ubuntu 14.04.1

### Scenario:
I'm trying to install several gems using gem_package.  I'm simply providing a gem name and the action of :install and options.  After reading the source code it appears that the "option" string I provide results in Chef shelling out to gem vs. using the API and I believe the shell out is broken.  Here's the simple loop:

```ruby
%w(pingdom-client pingdom-to-graphite).each do |the_gem|
  gem_package the_gem do
    action :install
    options '--source="http://gems.my.co"'
  end
end
```

### Expected Result:
The gems should just install.

### Actual Result:
Gems fail to install as the shell out is passing -v "" for version, which isn't valid

==> default: Mixlib::ShellOut::ShellCommandFailed
==> default:
==> default: ------------------------------------
==> default:
==> default: Expected process to exit with [0], but received '1'
==> default:     ---- Begin output of /usr/bin/gem install pingdom-client -q --no-rdoc --no-ri -v "" --source="http://gems.my.co" ----
==> default:
==> default: STDOUT:
==> default:     STDERR: ERROR:  While executing gem ... (ArgumentError)
==> default:
==> default:
==> default:     Illformed requirement [""]
==> default:
==> default: ---- End output of /usr/bin/gem install pingdom-client -q --no-rdoc --no-ri -v "" --source="http://gems.my.co" ----
==> default:
==> default: Ran /usr/bin/gem install pingdom-client -q --no-rdoc --no-ri -v "" --source="http://gems.my.co" returned 1
==> default:
==> default:

**Comment by lamont-granquist** [comment]
I might have fixed this in the multipackage patch because the specs trolled me and I fixed the `-v ""` stuff...  I might have been worried about backcompat and didn't fix it and preserved the broken behavior though because I couldn't make any sense out of it though...

**Comment by tas50** [comment]
I confirmed this is fixed in 12.5.1
